### PR TITLE
feat!: access (view/delete) recipe by index

### LINKED ISSUE #83: feat!: access recipes by index
As a CLI-user, although I can type fast, I want to be able to access a recipe by its index.

For example, I want to be able to delete/view/edit the first recipe by doing something like
`delete 1` instead of needing to type the full name of the recipe

### PR DESCRIPTION
**Pull request by yusufaine**
Closes #83.

This was a quick solution so it might not be *optimal*.
This can be used to for other index-based operations as well.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -21,13 +22,15 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the recipe identified by the name used in the displayed list of recipes.\n"
             + "Parameters: name (must be a valid case sensitive name)\n"
-            + "Example: " + COMMAND_WORD + " Aglio Olio";
+            + "Example: " + COMMAND_WORD + " Aglio Olio\n"
+            + "Example: " + COMMAND_WORD + " -i 3";
```
Comment (blaketingyu) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:26]: I don't think i have an issue with the usage of the tag "-i" here to specify index, just wondering if it's going to be a problem if "-i" is being used for ingredients as well.

Comment (yusufaine) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:26]: @blaketingyu I was considering that as well, but as the user, "-i" would only trigger the Ingredient Prefix if the command is an `AddCommand`.

**Review by blaketingyu — APPROVED** [review]
LGTM overall! 🌮 Just got that one comment.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (blaketingyu) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:26]: Yeap right,  i was thinking of having no prefix but at the same time, I don't mind having "-i" used for index as well.  Also, "-i" will trigger the ingredient prefix if the command is an `EditCommand` that we are implementing soon.

**Review by gigituang — APPROVED** [review]
LGTM! i guess we should keep to deleting by both name and index

**Pull request merged**